src/main/java/com/coyoapp/tinytask/service/DefaultFileService.java

```diff
@@ -0,0 +1,60 @@
+package com.coyoapp.tinytask.service;
+
+import com.coyoapp.tinytask.exception.FileException;
+import com.coyoapp.tinytask.repository.FileRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.Optional;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class DefaultFileService implements FileService {
+
```

Comment:
constructor injection should be preferred (see other components) instead of field injection

Reply:
Roger that!